### PR TITLE
FEATURE: Document environment and l10n path changes

### PR DESCRIPTION
**Pull request by DanielSiepmann**
* Add new section about Environment class and new paths.
* Adjust typo3conf/l10n occurrences to use new Environment class
  information.

Relates: #321

### CODE REVIEW

File: Documentation/ApiOverview/Environment/Index.rst [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+The fully qualified class name is :php:`\TYPO3\CMS\Core\Core\Environment`. The
+class provides static methods to access the necessary information.
+
+Also for testing, the :php:`initialize()` method can be called to adjust the
+information.
+
+.. _Environment-project-path:
+
+project path
```
Comment (mbrodala) [inline Documentation/ApiOverview/Environment/Index.rst]: Please use upper camel case or at least capitalization for headlines. Alternatively use the API method as headline.

File: Documentation/ApiOverview/Environment/Index.rst [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+------------
+
+The environment provides the path to the folder containing the
+:file:`composer.json`, or public web folder for non composer projects.
+
+The path is available through :php:`getProjectPath()` method.
```
Comment (mbrodala) [inline Documentation/ApiOverview/Environment/Index.rst]: I am missing a `the` everywhere. Also I'd always write `Environment::getFoo()` to make the usage obvious and even allow for copy and paste.

File: Documentation/ApiOverview/Environment/Index.rst [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+.. _Environment-var-path:
+
+var path
+--------
+
+The environment provides the path to :file:`var` folder. This folder contains
+data like logs, sessions, locks and cache files.
+
+The folder is, by default, :file:`typo3temp/var` for setups where the project
+path is equal to public path, e.g. "classic" installation without composer.
```
Comment (mbrodala) [inline Documentation/ApiOverview/Environment/Index.rst]: I'd stick with one variant, in other locations it's _non composer setups_.

File: Documentation/ApiOverview/Environment/Index.rst [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+Also for testing, the :php:`initialize()` method can be called to adjust the
+information.
+
+.. _Environment-project-path:
+
+project path
+------------
+
+The environment provides the path to the folder containing the
+:file:`composer.json`, or public web folder for non composer projects.
```
Comment (mbrodala) [inline Documentation/ApiOverview/Environment/Index.rst]: _Composer_ should be capitalized

File: Documentation/ApiOverview/Environment/Index.rst [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+var path
+--------
+
+The environment provides the path to :file:`var` folder. This folder contains
+data like logs, sessions, locks and cache files.
+
+The folder is, by default, :file:`typo3temp/var` for setups where the project
+path is equal to public path, e.g. "classic" installation without composer.
+
+For setups where project path and public path are not equal, the default is
```
Comment (mbrodala) [inline Documentation/ApiOverview/Environment/Index.rst]: Here I'd prefer `For Composer setups the default is ...`.

File: Documentation/ApiOverview/Environment/Index.rst [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+.. _Environment-config-path:
+
+config path
+-----------
+
+The environment provides the path to :file:`typo3conf`. This folder contains all
+TYPO3 global configuration files and folders, e.g. :file:`LocalConfiguration.php`.
+
+The folder is, by default, :file:`typo3conf` for setups where the project
+path is equal to public path, e.g. "classic" installation without composer.
```
Comment (mbrodala) [inline Documentation/ApiOverview/Environment/Index.rst]: Same here, we should stick with the established variants.

File: Documentation/ApiOverview/Environment/Index.rst [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+.. _Environment-labels-path:
+
+labels path
+-----------
+
+The environment provides the path to :file:`labels`, respective :file:`l10n`
+folder. This folder contains downloaded translation files.
+
+The folder is, by default, :file:`typo3temp/l10n` for setups where the project
+path is equal to public path, e.g. "classic" installation without composer.
```
Comment (mbrodala) [inline Documentation/ApiOverview/Environment/Index.rst]: You get the idea. ;-)

File: Documentation/ApiOverview/Environment/Index.rst (hunk 2 above)
Comment (DanielSiepmann) [reply Documentation/ApiOverview/Environment/Index.rst]: The only thing I didn't got is about `the`, what do you mean with

> I am missing a the everywhere.

Comment (mbrodala) [reply Documentation/ApiOverview/Environment/Index.rst]: It should be _The path is available through **the** `getProjectPath()` method._

**Pull request merged**